runtime/dart_snapshot.cc

```diff
@@ -23,8 +23,10 @@ const char* DartSnapshot::kIsolateInstructionsSymbol =
 // On Windows and Android (in debug mode) the engine finds the Dart snapshot
 // data through symbols that are statically linked into the executable.
 // On other platforms this data is obtained by a dynamic symbol lookup.
-#define DART_SNAPSHOT_STATIC_LINK \
-  (OS_WIN || (OS_ANDROID && FLUTTER_RUNTIME_MODE == FLUTTER_RUNTIME_MODE_DEBUG))
+#define DART_SNAPSHOT_STATIC_LINK                                        \
+  (OS_WIN ||                                                             \
+   (OS_ANDROID && (FLUTTER_RUNTIME_MODE == FLUTTER_RUNTIME_MODE_DEBUG || \
```

Comment:
I think it would be clearer to define a macro that is set whenever the engine is built with a JIT runtime.

It might also be useful to create a macro for modes that are "release modes".  I suspect there are other places where the engine currently checks for `FLUTTER_RUNTIME_MODE_RELEASE` that should be updated to include `JIT_RELEASE` (e.g. `shell/common/switches.cc`)

These can be defined by a new header in `flutter/common` based on the `FLUTTER_RUNTIME_MODE` set by the build scripts.

Reply:
Do you have any preference towards a positive/negative approach on these macros? For example PRECOMPILED_RUNTIME (which matches the dart_vm getter) vs JIT_RUNTIME (which calls out the unique behavior) ?

Reply:
I'd vote for `FLUTTER_JIT_RUNTIME` and `FLUTTER_RELEASE` as names for these macros

Reply:
I've updated them where appropriate to use `FLUTTER_RELEASE` and `FLUTTER_JIT_RUNTIME`